account-compression/sdk/src/events/index.ts

```diff
@@ -39,10 +46,79 @@ export function deserializeApplicationDataEvent(
     .toFixedFromData(data, 0)
     .read(data, 0);
   switch (event.__kind) {
-    case 'ApplicationData': {
+    case "ApplicationData": {
       return event.fields[0];
     }
     default:
-      throw Error('Unable to decode buffer as ApplicationDataEvent');
+      throw Error("Unable to decode buffer as ApplicationDataEvent");
   }
 }
+
+/**
+ * Helper function to extract the ChangeLogEvent V1 events from a TransactionResponse
+ * @param txResponse - TransactionResponse from the `@solana/web3.js`
+ * @param programId - PublicKey of the program (aka `programId`) that utilized the leaf on the tree
```

Comment:
Thank you for adding this documentation! This description is actually slightly misleading. This function only parses ChangeLogs from the **first** outer instruction that was sent to `programId`. If you want to get **all** changelogs for **all** invocations of the given `programId`, then need to add a loop over outerInstructions **and** also attempt to parse the inner instruction nesting 🫠